docs/docusaurus/docs/conceptual_guides/metricproviders.md

```diff
@@ -0,0 +1,105 @@
+
+- `column_values.in_set.unexpected_count`
+- `column_values.in_set.unexpected_rows`
+- `column_values.in_set.unexpected_values`
+- `column_values.in_set.unexpected_value_counts`
+
+To allow Expectations to work with multiple backends, methods for calculating Metrics need to be implemented for each ExecutionEngine. For example the `aggregate_fn` utility that is used with SQL and Spark requires the `column_values.in_set.unexpected_count.aggregate_fn` Metric. 
+
+Metrics can help you incorporate conditional statements in Expectations that support conditional evaluations. For example, `column_values.in_set.condition`.
+
+Metrics such as `column_values.in_set.unexpected_index_list` and `column_values.in_set.unexpected_index_query` can help you calculate the truthiness of your data.
+
+## Class hierarchy
+
+Although the class hierarchy for MetricProviders and Expectations is different, they use the same naming conventions. The following is the MetricProviders class hierarchy:
+
+```text
+MetricProvider
+    QueryMetricProvider
+    TableMetricProvider
+        ColumnAggregateExpectation
+    MapMetricProvider
+        ColumnMapMetricProvider
+            RegexColumnMapMetricProvider
+            SetColumnMapMetricProvide
+        ColumnPairMapMetricProvider
+        MulticolumnMapExpectation
+        MulticolumnMapMetricProvider
+```
+
+If you’re not sure which MetricProvider you should use with an Expectation, you can usually infer the correct MetricProvider class name from the Expectation class name. For example:
+
+| Expectation class                 | MetricProvider class                    |
+| --------------------------------- | --------------------------------------- |
+| Expectation                       | MetricProvider                          |
+| BatchExpectation                  | TableMetricProvider                     |
+| ColumnAggregateExpectation        | ColumnAggregateMetricProvider           |
+| ColumnMapExpectation              | ColumnMapMetricProvider                 |
+| RegexBasedColumnMapExpectation    | RegexColumnMapMetricProvider (built in) |
+| SetBasedColumnMapExpectation      | SetColumnMapMetricProvider (built in)   |
+| ColumnPairMapExpectation          | ColumnPairMapMetricProvider             |
+| MulticolumnMapExpectation         | MulticolumnMapMetricProvider            |
+| QueryExpectation                  | QueryMetricProvider                     |
+
+
+Sometimes, the MetricProvider class is created directly from the Expectation class, so you don’t need to specify a MetricProvider or methods when declaring a new Expectation. For example, the RegexColumnMapMetricProvider and the SetColumnMapMetricProvider classes.
```

Comment:
The example feels like a fragment.  Maybe have it be something like "For example, the RegexBasedColumnMapExpectation automatically implements the RegexColumnMapMetricProvider, and the SetBasedColumnMapExpectation automatically implements the SetColumnMapMetricProvider."

It would also be ideal to provide a link to a reference of which Expectations automatically implement a MetricProvider, but I'm not certain we have one currently.  It might be worth suggesting an update be made to Expectation class docstrings so that they indicate if a MetricProvider is automatically implemented with them, and then link to the Expectations section of the API docs in a future PR.

Reply:
@Rachel-Reverie I've implemented the text edit you recommended. Who could help with the Expectation class docstrings  you have suggested?